File: Data Structures & Algorithms/two-integer-sum/submission-10.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> remainders = new HashMap();
        for (int i = 0; i < nums.length; i++) {
            int curr = target - nums[i];
            if (remainders.containsKey(curr)) {
                return new int[]{remainders.get(curr), i};
            }
            remainders.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-10.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L, R = 0, len(numbers)-1
        while L < R:
            curr_sum = numbers[L] + numbers[R]
            if target < curr_sum:
                R -= 1
            elif target > curr_sum:
                L += 1
            else:
                return [L+1, R+1]